specs/interop/derivation.md

```diff
@@ -0,0 +1,52 @@
+# Derivation
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Deposit Context](#deposit-context)
+  - [Gas Considerations](#gas-considerations)
+- [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+This is an experimental section and may be changed in the future. It is not required
+for the initial release.
+
+### Deposit Context
+
+Derivation is extended to create **deposit contexts**, which signifies the execution of a depositing transaction.
+A deposit context is scoped to a single block, commencing with the execution of the first deposited transaction
+and concluding immediately after the execution of the final deposited transaction within that block.
+As such, there is exactly one deposit context per block.
+
+A deposit context is created by two operations:
+
+- An L1 attributes transaction that sets `isDeposit = true` on the `L1Block` contract.
+This instantiates a deposit context for the current block.
+- An L1 attributes transaction that sets `isDeposit = false`. This destroys the existing deposit context.
+
+These two operations wrap user deposits, such that `isDeposit = true` occurrs during the first L1 attributes
+transaction and `isDeposit = false` occurs immediately either after either the last user deposit,
+if one exists, or after the first L1 attributes transaction if there are no user deposits.
+
+The order of deposit transactions occurs as follows:
+
+1. L1 attributes transaction calling [`setL1BlockValuesInterop()`](./l1_attributes.md).
+1. User deposits
+1. L1 attributes transaction calling [`depositsComplete()`](./l1_attributes.md)
```

Comment:
```suggestion
1. L1 attributes transaction calling [`setL1BlockValuesInterop()`](./l1_attributes.md).
2. User deposits
3. L1 attributes transaction calling [`depositsComplete()`](./l1_attributes.md)
```

Reply:
oh, this works just fine, and we don't have to worry about re-ordering the numbers if we introduce an intermediary step

Reply:
ah ok, don't you think using this is better then?
```
* L1 attributes transaction calling [`setL1BlockValuesInterop()`](./l1_attributes.md).
* User deposits
1* L1 attributes transaction calling [`depositsComplete()`](./l1_attributes.md)
```